crypto/cpuid.c

```diff
@@ -126,23 +126,40 @@ void OPENSSL_cpuid_setup(void)
             vec = OPENSSL_ia32_cpuid(OPENSSL_ia32cap_P);
         }
 
-        if ((env = ossl_strchr(env, ':')) != NULL) {
-            IA32CAP vecx;
-
-            env++;
-            off = (env[0] == '~') ? 1 : 0;
-            vecx = ossl_strtouint64(env + off);
-            if (off) {
-                OPENSSL_ia32cap_P[2] &= ~(unsigned int)vecx;
-                OPENSSL_ia32cap_P[3] &= ~(unsigned int)(vecx >> 32);
-            } else {
-                OPENSSL_ia32cap_P[2] = (unsigned int)vecx;
-                OPENSSL_ia32cap_P[3] = (unsigned int)(vecx >> 32);
+        /* Processed indexes 0, 1 */
+        if ((env != NULL) && ((env = ossl_strchr(env, ':')) != NULL)) {
```

Comment:
The initial `env != NULL` here makes no sense. We have already derefed `env` in the code above, and confirmed that `env` is non NULL in the `ossl_get_env` call.

Reply:
removed extra check